test/test_sparse_csr.py

```diff
@@ -1409,6 +1409,20 @@ def test_exercise_detach(self, device, dtype):
             self.assertEqual(inp.crow_indices(), detached_inp.crow_indices())
             self.assertEqual(inp.col_indices(), detached_inp.col_indices())
 
+    @skipMeta
+    @dtypes(*get_all_dtypes(include_bool=False, include_bfloat16=False, include_complex=False))
+    def test_eye(self, device, dtype):
+        for n in (3, 5, 7):
+            # This will fail with an error:
+            # RuntimeError: torch.empty: Only 2D sparse CSR tensors are supported.
+            sparse_csr_output = torch.eye(n, layout=torch.sparse_csr, device=device, dtype=dtype)
```

Comment:
It works fine when a sparse CSR tensor is passed as `out=` argument, but fails when the layout is passed.

```pyhon
x = torch.tensor([
    [1, 0, 2],
    [2, 0, 1],
    [0, 0, 3]
]).to_sparse_csr()
torch.eye(3, out=x)
print('x: ', x)
```

Output:

```python
x:  tensor(crow_indices=tensor([0, 1, 2, 3]),
       col_indices=tensor([0, 1, 2]),
       values=tensor([1, 1, 1]), size=(3, 3), nnz=3, layout=torch.sparse_csr)
```

@IvanYashchuk - would you have any idea on this?

Reply:
Update: attempting to implement functional version instead.

Reply:
In the implementation of `eye` functional variant, `torch.empty({0}, options);` is called. Probably branching based on the given layout can help. Working on it.

Reply:
Fixed.